src/Graph.ts

```diff
@@ -73,3 +80,85 @@ function isAnExit(board: Board, x: number, y: number) {
     y === lastRowIndex
   );
 }
+
+export function positionToCoordinate(position: string): {
+  x: number;
+  y: number;
+} {
+  if (position.length > 3) {
+    return { x: -1, y: -1 };
+  }
+
+  const positionSplit = position.split("");
+
+  const x = ALPHABET.indexOf(positionSplit[0]);
+  const y = parseInt(positionSplit[1] + positionSplit[2]);
+
+  if (!x || x === -1 || !y) {
+    return { x: -1, y: -1 };
+  }
+
+  return { x, y };
+}
+
+function nextCoordinateFromDirection(
+  coordinate: { x: number; y: number },
+  direction: string
+): { x: number; y: number } {
+  if (!coordinate || !direction) {
+    return { x: -1, y: -1 };
+  }
+
+  if (direction == "W" || direction == "E") {
+    return { x: coordinate.x + VALUE_DIRECTION[direction], y: coordinate.y };
+  }
+
+  return { x: coordinate.x, y: coordinate.y + VALUE_DIRECTION[direction] };
+}
+
+export function moveMarbleInDirection(
+  graph: Graph,
+  marbleCoordinate: { x: number; y: number },
+  direction: string
+): Graph {
+  if (!marbleCoordinate || !graph || !direction) {
+    return {
+      nodes: {},
+      edges: [],
+    };
+  }
+
+  const firstNode = graph.nodes[`${marbleCoordinate.y},${marbleCoordinate.x}`];
+
+  if (firstNode.value === 0) {
+    return graph;
+  }
+
+  const nodes = [firstNode];
+  while (nodes) {
```

Comment:
This is already equals to true and will always be true, so `while (true)` seems to be more appropriate.

However take care of having while(true) in a codebase, you have to test each cases, like boundaries, of your board to be sure you cannot encounter an infinite loop

Reply:
A recursive function could have replaced this while. I am not familiar with it, but I note it.